lib/agent.js

```diff
@@ -517,8 +517,24 @@ Agent.prototype._fetchOps = function(collection, id, version, callback) {
 Agent.prototype._fetchBulk = function(collection, versions, callback) {
   if (Array.isArray(versions)) {
     this.backend.fetchBulk(this, collection, versions, function(err, snapshotMap) {
-      if (err) return callback(err);
-      callback(null, {data: getMapData(snapshotMap)});
+      var isPartialError = err != null && err.code === ERROR_CODE.ERR_SNAPSHOT_READS_REJECTED;
```

Comment:
I don't think a partial error should produce an error in this callback. Could we simply have the error information in the snapshotMap? I wonder if we even need to change this code in agent if we do that

Reply:
Yes, it's possible, though I did have to update the agent code to serialize errors from the new location.